MOBILE_CLIENT/COMMON/_HIGH/XPATH_INJECTION/recommendation.md

```diff
@@ -0,0 +1,91 @@
+
+# Xpath Injection
+
+To mitigate XPath Injection vulnerabilities, it is important to properly validate and sanitize user input before using it in XPath queries. This can be done by using parameterized queries or prepared statements, which separate the user input from the query logic. Additionally, limiting the privileges of the user executing the query can help prevent unauthorized access to sensitive data. Regularly updating and patching software and libraries can also help prevent known vulnerabilities from being exploited. Finally, implementing logging and monitoring can help detect and respond to any attempted attacks.
+
+# Code Examples:
+
+### Dart
+
+```dart
+
+bool _validate_query(String _searchQuery){
+  // check for special characters
+  for(var i = 0; i < tokens.length; i++){
+      if (string.contains(new RegExp(r'[A-Z]')) == false){
+        return false;
+      }
+    }
+  return true;
+}
+
+void _fetch_data(String _searchQuery) {
+  
+  // validate user input
+  if (_validate_query(_searchQuery) == false){
+    // raise error
+    return ;
+  }
+  final content = XmlDocument.parse(xmlFileContent);
+  final xml_node = XmlXPath.node(content);
+  final xpath = xml_node.query('//book[author=$_searchQuery');
+
+  showDialog(
+    context: context,
+    builder: (context) => AlertDialog(
+      title: Text('Search Result'),
+      content: Text(result),
+    ),
+  );
+  }
+```
+
+### Swift
+
+```swift
+import Foundation
+import SWXMLHash
+
+func main() {
+    print("Enter search term:")
+    guard let searchTerm = readLine()?.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else {
+        print("Invalid search term")
+        return
+    }
+    
+    let xml = SWXMLHash.parse(xmlString)
+    let results = xml["books"]["book"].all(withAttribute: "title", matchingXPath: "//title[contains(text(), '\(searchTerm)')]")
+    
+    for result in results {
+        print(result["title"].element!.text)
+        print(result["author"].element!.text)
+    }
+}
+```
+
+### Kotlin
+
+```kotlin
+import javax.xml.parsers.DocumentBuilderFactory
+import org.xml.sax.InputSource
+import java.io.StringReader
+
+fun main() {
+    val userInput = readLine() ?: ""
```

Comment:
I don't see the issue here.

Reply:
[d270584](https://github.com/Ostorlab/KB/pull/58/commits/d27058446edcfd6aedc081445ea756617e2dec5a), please double check the new snippet, also this one uses parameterized query.